frontend/views/containers/proposals/AddMembers.vue

```diff
@@ -61,33 +86,141 @@ export default {
+        const userState = await sbp('state/latestContractState', contractID)
+        this.ephemeral.invitesToSend[index] = { contractID, ...userState }
+        Vue.set(this.form.eachFeedbackMsg, index, {
+          text: this.L('Ready to be invited!'),
+          class: 'has-text-success'
+        })
+        this.ephemeral.isValid = true
+      } catch (err) {
+        console.log(err)
+        this.setInviteError(index, this.L('Non existing user. TODO support general invites'))
+      }
+    },
+    setInviteError (index, text) {
+      this.ephemeral.invitesToSend[index] = null
+      Vue.set(this.form.eachFeedbackMsg, index, {
+        text,
+        class: 'error'
+      })
+    },
+    removeInvitee (index) {
+      this.ephemeral.invitesCount -= 1
+      this.form.invitees.splice(index, 1)
+      this.ephemeral.invitesToSend[index] = null
+    },
+    addInviteeSlot () {
+      this.ephemeral.invitesCount += 1
+    },
+    async submit (form) {
+      try {
+        const groupId = this.currentGroupId
+        const groupName = this.groupSettings.groupName
+        // NOTE: All invitees proposals will expire at the exact same time.
+        // That plus the proposal creator is what we'll use to know
+        // which proposals should be displayed visually together.
+        const expiresDateMs = Date.now() + this.groupSettings.proposals[PROPOSAL_INVITE_MEMBER].expires_ms
+        for (const index in this.ephemeral.invitesToSend) {
+          const member = this.ephemeral.invitesToSend[index]
+          const mailbox = member.attributes.mailbox
+          const memberName = member.attributes.name
+
+          if (this.groupShouldPropose) {
+            const proposal = await sbp('gi.contracts/group/proposal/create',
+              {
+                proposalType: PROPOSAL_INVITE_MEMBER,
+                proposalData: {
+                  member: memberName,
+                  reason: form.reason
+                },
+                votingRule: this.groupSettings.proposals[PROPOSAL_INVITE_MEMBER].rule,
+                expires_date_ms: expiresDateMs
+              },
+              groupId
+            )
+            await sbp('backend/publishLogEntry', proposal)
+
+            Vue.set(this.form.eachFeedbackMsg, index, {
+              text: this.L('Member proposed successfully!'),
+              class: 'has-text-success'
+            })
+          } else {
+            const invite = generateInvites(1)
+            const inviteToMailbox = await sbp('gi.contracts/mailbox/postMessage/create',
+              {
+                messageType: TYPE_MESSAGE,
+                from: groupName,
+                subject: `You've been invited to join ${groupName}!`,
+                message: `Hi ${memberName},
+
+                Here's your special invite link:
+
+                ${buildInvitationUrl(this.$store.state.currentGroupId, invite.inviteSecret)}`
+              },
+              mailbox
+            )
```

Comment:
Yeah... based on @mmbotelho's comment, we really don't need to use the mailbox anymore. So both the username validation code, and the mailbox sending code, can be removed...